AsyncDisplayKit/ASCollectionNode.mm

```diff
@@ -273,6 +317,20 @@ - (void)setTuningParameters:(ASRangeTuningParameters)tuningParameters forRangeMo
   return [self.rangeController setTuningParameters:tuningParameters forRangeMode:rangeMode rangeType:rangeType];
 }
 
+#pragma mark - Selection
+
+- (void)selectItemAtIndexPath:(nullable NSIndexPath *)indexPath animated:(BOOL)animated scrollPosition:(UICollectionViewScrollPosition)scrollPosition
+{
+  ASDisplayNodeAssert([self isNodeLoaded], @"ASCollectionNode should be loaded before calling selectItemAtIndexPath");
```

Comment:
I agree, I added a todo referring to your discussion post.

I added the `ASDisplayNodeAssertMainThread()`, I didn't add it before because the code that this method calls checks for that assertion. But I suppose it's better to have it in case the implementation changes later on.